lib/client/config.go

```diff
@@ -254,15 +256,65 @@ func (c *config) GetNodeLogLevel(node string) (string, ConfigLocation, error) {
 }
 
 // GetFelixConfig provides a mechanism for getting arbitrary Felix configuration
-// in the datastore.
-func (c *config) GetFelixConfig(name string) (*string, error) {
-	return c.getValue(model.GlobalConfigKey{Name: name})
+// in the datastore.  An empty value for the node will get the global
+// configuration, a non-empty node will get the per-node configuration.
+func (c *config) GetFelixConfig(name, node string) (*string, error) {
+	if node == "" {
+		return c.getValue(model.GlobalConfigKey{Name: name})
+	} else {
+		return c.getValue(model.HostConfigKey{Hostname: node, Name: name})
+	}
 }
 
 // SetFelixConfig provides a mechanism for setting arbitrary Felix configuration
 // in the datastore.  A nil value will remove the entry from the datastore.
-func (c *config) SetFelixConfig(name string, value *string) error {
-	key := model.GlobalConfigKey{Name: name}
+// An empty value for the node will set the global configuration, a non-empty
+// node will set the per-node configuration.
+//
+// Caution should be observed using this method as no data validation is
+// performed, and setting incorrect values may have unexpected consequences.
+func (c *config) SetFelixConfig(name, node string, value *string) error {
+	var key model.Key
+	if node == "" {
+		key = model.GlobalConfigKey{Name: name}
+	} else {
+		key = model.HostConfigKey{Hostname: node, Name: name}
+	}
+	if value == nil {
```

Comment:
umm when val==nil we are deleting using the key .... so I can't move this to the top.